src/core/commandVersionUpgrades/upgradeV3ToV4/upgradeV3ToV4.ts

```diff
@@ -0,0 +1,56 @@
+import { CommandV4 } from "../../commandRunner/typings/CommandV4.types";
+import { CommandV3 } from "../../commandRunner/typings/legacy/CommandV3.types";
+import {
+  PartialPrimitiveTargetDescriptorV3,
+  PartialTargetDescriptorV3,
+} from "../../commandRunner/typings/legacy/PartialTargetDescriptorV3.types";
+import {
+  ImplicitTargetDescriptor,
+  PartialPrimitiveTargetDescriptor,
+  PartialRangeTargetDescriptor,
+  PartialTargetDescriptor,
+} from "../../commandRunner/typings/PartialTargetDescriptor.types";
+
+export function upgradeV3ToV4(command: CommandV3): CommandV4 {
+  return {
+    ...command,
+    version: 4,
+    targets: command.targets.map(upgradeTarget),
+  };
+}
+
+function upgradeTarget(
+  target: PartialTargetDescriptorV3,
+): PartialTargetDescriptor {
+  switch (target.type) {
+    case "primitive":
+      return upgradePrimitiveTarget(target);
+    case "range": {
+      const { anchor, ...rest } = target;
+      return {
+        ...rest,
+        anchor: upgradePrimitiveTarget(anchor),
+      };
+    }
+    case "list": {
+      const { elements, ...rest } = target;
+
+      return {
+        ...rest,
+        elements: elements.map(upgradeTarget) as [
+          PartialPrimitiveTargetDescriptor | PartialRangeTargetDescriptor,
+        ],
```

Comment:
This should be an array rather than a 1-tuple, right?

Reply:
Nice catch 😊